### PR TITLE
Update icewind/streams and icewind/smb

### PR DESCRIPTION
**Pull request by icewind1991**
requires: https://github.com/owncloud/3rdparty/pull/260

smb update consists of a minor fix for stream_stat and changing it's composer.json to allow an updated icewind/streams to be installed

streams update has `RetryWrapper` which is needed to make encryption work consistently with remote streams

cc @PVince81 @schiesbn @LukasReschke

### CODE REVIEW

**Comment by PVince81** [comment]
@icewind1991 I think we'll also need https://github.com/icewind1991/SMB/pull/46

**Comment by schiesbn** [comment]
:+1:

**Comment by icewind1991** [comment]
> @icewind1991 I think we'll also need icewind1991/SMB#46

Will add

**Comment by icewind1991** [comment]
done

File: apps/files_external/3rdparty/icewind/streams/src/CallbackWrapper.php [hunk 1]
```diff
@@ -39,54 +40,56 @@ class CallbackWrapper extends Wrapper {
...
 	 * @param callable $read (optional)
 	 * @param callable $write (optional)
 	 * @param callable $close (optional)
+	 * @param callable $readDir (optional)
 	 * @return resource
 	 *
 	 * @throws \BadMethodCallException
 	 */
-	public static function wrap($source, $read = null, $write = null, $close = null) {
+	public static function wrap($source, $read = null, $write = null, $close = null, $readDir = null) {
```
Comment (PVince81) [inline apps/files_external/3rdparty/icewind/streams/src/CallbackWrapper.php]: Hmmm, I hope we won't backport these too?

**Comment by PVince81** [comment]
:+1:

**Comment by PVince81** [comment]
https://github.com/owncloud/3rdparty/pull/260 was merged.

@icewind1991 please update this branch's submodule to match

**Comment by PVince81** [comment]
I have updated the submodule to save some time

**Comment by PVince81** [comment]
@karlitschek backport to 9.0.1 ? We need the RetryWrapper to fix FTP + encryption and the SMB update for DFS: https://github.com/owncloud/core/issues/22286#issuecomment-197285301

**Comment by PVince81** [comment]
stable9: https://github.com/owncloud/core/pull/23359

**Comment by karlitschek** [comment]
great. please backport :+1:

**Comment by DeepDiver1975** [comment]
@PVince81 has conflicts - THX

**Comment by PVince81** [comment]
@icewind1991 please resolve conflict

**Comment by icewind1991** [comment]
rebased

**Pull request merged**